doc/TransmissionProtocol/TransmissionProtocol.tex

```diff
@@ -0,0 +1,354 @@
+to reduce our chances of collisions.
+
+The iCRAB protocol is expected to support only a finite number of transmitter
+UIDs.
+The upper bound selected is fairly arbitrary but based on the client's expected
+needs for the project's research.
+
+Finally, this protocol is also expected to be able to encode (along with the
+ID of a given transmitter) a boolean value.
+For the purposes of the project, this value will be \texttt{true}
+if and only if a given  transmitter is inert.
+This generally indicates that the crab died or molted its shell.
+We will encode this boolean value by effectively doubling the number of IDs
+that can be encoded.
+Half of the IDs are to be used when the boolean value is \texttt{false},
+and the other half are used when it is \texttt{true}.
+Specifically, values in the range $[0, MAX\_UID)$ are to be used when
+\texttt{inert=false}.
+There is no additional mapping when IDs are in this range.
+For all values in the range $[MAX\_UID, 2*MAX\_UID)$, the value encoded is
+actually $MAX\_UID$ bigger than the ID that is encoded, and
+\texttt{inert=true}.
+
+For example, if a signal encodes the value 42, the receiver should report
+this as transmitter 42 with \texttt{inert=false}.
+However, if the value 542 is received, assuming that $MAX\_UID=500$,
+the receiver should interpret this as transmitter 42 but with
+\texttt{inert=true}.
+
+For more background on the topic of boolean value encoding, see RFC 2.
+
+\subsection{Preamble to Definitions and Specifications} \label{preamble}
+
+Each Unique Transmission Pattern (UTP) will be formed by a ping, a delay,
+and another ping, in that order.
+Each transmitter will broadcast a UTP and will then wait for some interval
+before transmitting a UTP again.
+This process loops indefinitely throughout the transmitter's lifetime.
+
+Defined in Table \ref{table:constants} are the constants that we will use for
+the various aspects of this protocol.
+This section of the document will be updated as needed if these values
+change.
+
+Table \ref{table:formulae} lists the various mathematical formulae we will
+use for encoding.
+Some functions are passed a single integer value, which is a UID.
+
+Between each UTP broadcast, every transmitter should wait for some amount
+of time.
+As mentioned previously, this interval should include some random variation
+that is recalculated each time.
+The length of each interval should be a number of seconds in the range\\
+$[MIN\_INTERVAL,\ MAX\_INTERVAL]$.
+
+Finally, we formally define the behavior of a transmitter in pseudo-code
+(see Listing \ref{lst:tx}.)
+The functions {\em HIGH} and {\em LOW} cause the physical transmitter to begin
+or cease transmitting, respectively.
+The {\em sleep()} function simply causes execution of the code to stop for a
+given number of milliseconds.
+
+Section \ref{specs}, on the following page,
+aims to be the developer's one-stop shop for relevant constants,
```

Comment:
I suggest not using "one-stop shop" it sounds imprecise and colloquial in an otherwise very well articulated document.